packages/logger/README.md

```diff
@@ -0,0 +1,99 @@
+# Usage
+
+```js
+var Logger = require('@mongosh/logger')
+var path = require('path')
+var fs = require('fs')
+var os = require('os')
+
+var saveFilePath = path.join(os.homedir(), '.mongosh_log')
+var log = new Logger(saveFilePath)
+
+var infoObject = {
+  shellCommand: 'db.sales.find({ storeLocation: "Berlin" })'
+}
+
+var errorObject = {
+  shellCommand: 'db.sales.find({ storeLocation: "Berlin" })'
+  error: new TypeError()
+}
+
+log.info(infoObject)
+log.error(errorObject)
+```
+
+Logger uses [pino](https://github.com/pinojs/pino) under the hood for logging.
+
+Within mongosh architecture, `Logger` gets initialised along with the `Mapper`
+and requires passing in logger parameters to ShellMapper:
+
+![Using Logger with browser, Compass and CLI interfaces requires passing in
+logger parameters to ShellMapper](./logger-usage.png)
+
+```js
+var CliServiceProvider = require('mongosh-service-provider-server').CliServiceProvider
+var ShellApi = require('mongosh-shell-api')
+var Mapper = require('mongosh-mapper')
+
+var loggerPath = path.join(os.homedir(), '.mongosh_log')
+
+CliServiceProvider.connect(driverUri, driverOptions)
+  .then(function (serviceProvider) {
+    var mapper = new Mapper(serviceProvider, loggerPath)
+    var shellApi = new ShellApi(this.mapper)
+  })
+```
+
+## API
+### `new Logger([destination])`
+Create a new Logger instance. Provide destination path to file if you want to save
+information locally. Otherwise data will only be sent to
```

Comment:
Would never expect the logger to report info to DWH. 

I like that the telemetry / reporting implementation is passed as a dependency like the logger is. But by any mean i would segregate the two interfaces.

```
interface TelemetryReporter {
   reportEvent(type: string, payload: any): Promise<void>
}

interface Logger {
   info(...args: any): void;
   error(...args: any): void;
   warn(...args: any): void;
}

const segmentTelemetryReporter = new SegmentTelemetryReporter(segmentApiKey);
const fileLogger = require('pino')( pino.destination('/log/path'));

const mapper = new Mapper(shellApi, {logger: fileLogger, telemetryReporter: segmentTelemetryReporter});
```


```

Telemetry.reportEvent

Reply:
This RFC is for a logger that is a wrapper around both a logger and telemetry. Otherwise I'd straight up use `pino`. 

I don't think I understand this sentence:
> But by any mean i would segregate the two interfaces.

Reply:
Maybe a drawing will help on what this RFC is trying to encapsulate:

<img width="1024" alt="0B1F9CE1-E965-425E-BF81-2647449A40A3" src="https://user-images.githubusercontent.com/8107784/73848265-17002a80-4828-11ea-9604-08be66cf8829.png">

Reply:
Sorry for the confusion, I was referring to the interface segregation in the SOLID principles.